### PR TITLE
Add missing HB callback for DIM

### PR DESCRIPTION
**Pull request by pasitlaothamatas**
### Summary

- Modify heartbeat monitor to support DIM callback when HB is not detected
- Add board IDs. Refer to `App_SharedHeartbeatMonitor.h` to see how these are currently implemented

### Testing Done

- add #define for NDEBUG, test that we transition into the fault state and that 888 is displayed on the sseg display

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/shared/Src/App/App_SharedHeartbeatMonitor.c [hunk 1]
```diff
@@ -79,3 +87,18 @@ void App_SharedHeartbeatMonitor_CheckIn(
 {
     heartbeat_monitor->heartbeats_checked_in |= heartbeat;
 }
+
+#if (BOARD_ID == DIM_ID)
```
Comment (kirons96) [inline boards/shared/Src/App/App_SharedHeartbeatMonitor.c:91]: this indicates to me that this should be in a component specific file i.e. DIM file, maybe with a weak function or something

Comment (pasitlaothamatas) [reply boards/shared/Src/App/App_SharedHeartbeatMonitor.c:91]: I think using a function pointer here might not be the best way to invoke your callback

Comment (pasitlaothamatas) [reply boards/shared/Src/App/App_SharedHeartbeatMonitor.c:91]: Might be much easier to read if we had the heartbeat tick function return false when a heartbeat signal is not received. We can implement the "callback" in the state machine

Comment (pasitlaothamatas) [reply boards/shared/Src/App/App_SharedHeartbeatMonitor.c:91]: Rlly wanna be careful with fxn ptrs for next year and make sure we use it only when its needed. Makes the code rlly hard to follow tbh

**Comment by pasitlaothamatas** [comment]
image
